src/sagemaker/pytorch/README.rst

```diff
@@ -0,0 +1,711 @@
+
+=======================================
+SageMaker PyTorch Estimators and Models
+=======================================
+
+With PyTorch Estimators and Models, you can train and host PyTorch models on Amazon SageMaker.
+
+Supported versions of PyTorch: ``0.4.0``.
+
+You can visit the PyTorch repository at https://github.com/pytorch/pytorch.
+
+Table of Contents
+-----------------
+
+1. `Training with PyTorch <#training-with-pytorch>`__
+2. `PyTorch Estimators <#pytorch-estimators>`__
+3. `Distributed PyTorch Training <#distributed-pytorch-training>`__
+4. `Saving models <#saving-models>`__
+5. `Deploying PyTorch Models <#deploying-pytorch-models>`__
+6. `SageMaker PyTorch Model Server <#sagemaker-pytorch-model-server>`__
+7. `Working with Existing Model Data and Training Jobs <#working-with-existing-model-data-and-training-jobs>`__
+8. `PyTorch Training Examples <#pytorch-training-examples>`__
+9. `SageMaker PyTorch Docker Containers <#sagemaker-pytorch-docker-containers>`__
+
+
+Training with PyTorch
+------------------------
+
+Training PyTorch models using ``PyTorch`` Estimators is a two-step process:
+
+1. Prepare a PyTorch script to run on SageMaker
+2. Run this script on SageMaker via a ``PyTorch`` Estimator.
+
+
+First, you prepare your training script, then second, you run this on SageMaker via a ``PyTorch`` Estimator.
+You should prepare your script in a separate source file than the notebook, terminal session, or source file you're
+using to submit the script to SageMaker via a ``PyTorch`` Estimator. This will be discussed in further detail below.
+
+Suppose that you already have a PyTorch training script called `pytorch-train.py`.
+You can then setup a ``PyTorch`` Estimator with keyword arguments to point to this script and define how SageMaker runs it:
+
+```python
+
+    from sagemaker.pytorch import PyTorch
+
+    pytorch_estimator = PyTorch(entry_point="pytorch-train.py",
+                                role="SageMakerRole",
+                                train_instance_type="ml.p3.2xlarge",
+                                train_instance_count=1)
+```
+
+After that, you simply tell the estimator to start a training job and provide an S3 URL
+that is the path to your training data within Amazon S3:
+
+```python
+
+    pytorch_estimator.fit("s3://bucket/path/to/training/data")
+
+```
+
+In the following sections, we'll discuss how to prepare a training script for execution on SageMaker,
+then how to run that script on SageMaker using a ``PyTorch`` Estimator.
+
+
+Preparing the PyTorch Training Script
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Your PyTorch training script must be a Python 2.7 or 3.5 compatible source file.
+
+The training script is very similar to a training script you might run outside of SageMaker, but you
+can access useful properties about the training environment through various environment variables, such as
```

Comment:
Saying "such as" kind of makes me think there is a complete list somewhere of all of these variables, but I can't find it. How does a customer get a list of all the environment variables?

Reply:
It's bellow:

For more on training environment variables, please visit `SageMaker Containers <https://github.com/aws/sagemaker-containers>`_.